File: Euclid.Storage/Database/Access/SettingDao.cs
using Euclid.Storage.Database.Data;
using System.Collections.Generic;
using System.Linq;

namespace Euclid.Storage.Database.Access
{
    public class SettingDao
    {
        public static void GetSettings(out Dictionary<string, string> settings)
        {
            using (var session = SessionFactoryBuilder.Instance.SessionFactory.OpenSession())
            {
                settings = session.QueryOver<SettingsData>().List().ToDictionary(x => x.Key, x => x.Value);
            }
        }

        public static bool HasSetting(string key)
        {
            using (var session = SessionFactoryBuilder.Instance.SessionFactory.OpenSession())
            {
                return session.QueryOver<SettingsData>().Where(x => x.Key == key).List().Count > 0;
            }
        }

        public static void SaveSetting(string key, string value)
        {
            using (var session = SessionFactoryBuilder.Instance.SessionFactory.OpenSession())
            {
                using (var transaction = session.BeginTransaction())
                {
                    try
                    {
                        session.Save(new SettingsData { Key = key, Value = value });
                        transaction.Commit();
                    }
                    catch
                    {
                        transaction.Rollback();
                    }
                }
            }
        }
    }
}

File: euclid-storage/src/main/java/euclid/storage/database/access/SettingDao.java
package euclid.storage.database.access;

import euclid.storage.database.SessionFactoryBuilder;
import euclid.storage.database.data.SettingsData;
import org.hibernate.Session;
import org.hibernate.query.Query;

import java.util.LinkedHashMap;
import java.util.List;
import java.util.Map;

public class SettingDao {

    public static Map<String, String> getSettings() {
        try (Session session = SessionFactoryBuilder.getInstance().getSessionFactory().openSession()) {
            Query<SettingsData> q = session.createQuery("FROM SettingsData", SettingsData.class);
            List<SettingsData> list = q.getResultList();
            Map<String, String> settings = new LinkedHashMap<>();
            for (SettingsData s : list) {
                settings.put(s.getKey(), s.getValue());
            }
            return settings;
        }
    }

    public static boolean hasSetting(String key) {
        try (Session session = SessionFactoryBuilder.getInstance().getSessionFactory().openSession()) {
            Query<Long> q = session.createQuery(
                    "SELECT COUNT(s) FROM SettingsData s WHERE s.key = :key", Long.class);
            q.setParameter("key", key);
            return q.getSingleResult() > 0;
        }
    }

    public static void saveSetting(String key, String value) {
        try (Session session = SessionFactoryBuilder.getInstance().getSessionFactory().openSession()) {
            var transaction = session.beginTransaction();
            try {
                session.persist(new SettingsData(key, value));
                transaction.commit();
            } catch (Exception e) {
                if (transaction.isActive()) {
                    transaction.rollback();
                }
            }
        }
    }
}
